### ISSUE TITLE
EAS build: Failed to resolve plugin for module

### ISSUE THREAD

**Issue by Y-Smirnov** [expo/eas-cli#2696]
### Summary

Hello Expo Team,

EAS build fails to resolve a plugin inside a library.
This was also an issue in Expo and has been fixed in Expo 52 (more specifically @expo/config-plugins v9)
Links: [here](https://github.com/expo/expo/pull/31569) and [here](https://github.com/expo/expo/commit/4b75917d8456d8852bf95d17add90b464fa5fd62#diff-1de22228381392e614108ecfbf08f4a516e718cd9b736af5a44cbed8a4fd388cR21)

Although the app uses Expo v52 (which depends on config-plugins v9), eas-cli depends on @expo/config-plugins v8 (which seems to be used when doing eas build).
If that's the case, do you have any plans to update?

### Managed or bare?

Bare

### Environment

  expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 15.0
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
      Yarn: 1.22.22 - ~/.nvm/versions/node/v20.11.1/bin/yarn
      npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm
      Watchman: 2024.09.09.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.16.1 - /Users/yevheniismirnov/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
      Android SDK:
        API Levels: 28, 30, 31, 32, 33, 34, 35
        Build Tools: 29.0.2, 30.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.1, 33.0.2, 34.0.0, 35.0.0
        System Images: android-24 | Google APIs ARM 64 v8a, android-25 | Google APIs ARM 64 v8a, android-26 | Google APIs Intel x86 Atom_64, android-27 | Google APIs ARM 64 v8a, android-27 | Google Play Intel x86 Atom, android-28 | Google ARM64-V8a Play ARM 64 v8a, android-29 | Google APIs ARM 64 v8a, android-29 | Google Play ARM 64 v8a, android-30 | ARM 64 v8a, android-30 | Google APIs ARM 64 v8a, android-30 | Google Play ARM 64 v8a, android-31 | Google APIs ARM 64 v8a, android-33 | Google APIs ARM 64 v8a, android-33 | Google Play ARM 64 v8a, android-34 | Google APIs ARM 64 v8a, android-34 | Google Play ARM 64 v8a, android-35 | Google Play ARM 64 v8a
    IDEs:
      Android Studio: 2024.2 AI-242.21829.142.2421.12409432
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 13.2.0
    Expo Workflow: bare

### Error output

Failed to resolve plugin for module "react-native-*" relative to "/home/expo/workingdir/build/packages/sample-expo"

### Reproducible demo or steps to reproduce from a blank project

.

**Comment by szdziedzic** [comment]
Hey, we are working on it

**Comment by Y-Smirnov** [comment]
Hello @szdziedzic 

Thank you for taking care of the issue.
I did update to the latest eas-cli (eas-cli/13.4.2) but the issue persists.

Here is the error from the eas build:
```
Package "react-native-*" does not contain a valid config plugin.
Learn more: https://docs.expo.dev/guides/config-plugins/#creating-a-plugin

Cannot use import statement outside a module
```

I also did run _eas build_ with _local_ option (--local) and had the same issue.
Locally it seems the issue is related to the **eas-cli-local-build-plugin** being used for the build, which uses **@expo/config-plugins v8**.
On the EAS Cloud it seems the process may be similar (as it's referenced in [docs](https://docs.expo.dev/build-reference/local-builds/) (--local) simulates the build done on the CLoud) or maybe some other update is required (just sharing my point of view, I definitely don't have the whole picture as I would like to).

Also, I made sure to isolate the issue and ensure (again) the issue is related to have multiple **package.json** present.
Test scenario was:
- App uses library build normally (FAILS)
- App uses library build normally & removed "extra" **package.json** files during the build script (SUCCESS)
